Gradle 9.2.0 fails to start on `arm32v7`, `ppc64le` and `s390x`

### Current Behavior

```bash
# gradle --stacktrace --debug --version
# ...
FAILURE: Build failed with an exception.
* What went wrong:
java.lang.UnsupportedOperationException (no error message)
* Try:
> Run with --scan to generate a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.
* Exception is:
java.lang.UnsupportedOperationException
  at net.rubygrapefruit.platform.internal.Platform$Unsupported.getId(Platform.java:384)
  at org.gradle.internal.logging.sink.ConsoleConfigureAction.configureAutoConsole(ConsoleConfigureAction.java:70)
  at org.gradle.internal.logging.sink.ConsoleConfigureAction.execute(ConsoleConfigureAction.java:48)
  at org.gradle.internal.logging.sink.ConsoleConfigureAction.execute(ConsoleConfigureAction.java:43)
  at org.gradle.internal.logging.sink.OutputEventRenderer.attachProcessConsole(OutputEventRenderer.java:152)
  at org.gradle.internal.logging.services.DefaultLoggingManager$ProcessConsoleAttachment.run(DefaultLoggingManager.java:391)
  at org.gradle.internal.logging.services.DefaultLoggingManager$StartableLoggingRouter.addConsoleAttachement(DefaultLoggingManager.java:269)
  at org.gradle.internal.logging.services.DefaultLoggingManager$StartableLoggingRouter.attachProcessConsole(DefaultLoggingManager.java:275)
  at org.gradle.internal.logging.services.DefaultLoggingManager.attachProcessConsole(DefaultLoggingManager.java:216)
  at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:49)
  at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:27)
  at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:41)
  at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:26)
  at org.gradle.launcher.cli.DefaultCommandLineActionFactory$WithLogging.execute(DefaultCommandLineActionFactory.java:489)
  at org.gradle.launcher.Main.doAction(Main.java:35)
  at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:53)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:569)
  at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:72)
  at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:39)
  at org.gradle.launcher.GradleMain.main(GradleMain.java:26)
```

### Expected Behavior

Gradle works like in previous versions

### Context (optional)

https://github.com/docker-library/official-images/pull/20181#issuecomment-3463371500

https://github.com/docker-library/official-images/pull/20181#issuecomment-3470590238

### Self-contained Reproducer Project

`gradle --stacktrace --debug --version` **on actual hardware of the relevant architecture** 

### Gradle version

9.2.0

### Gradle version that used to work

9.1.0

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

--- a/platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/sink/ConsoleConfigureAction.java
+++ b/platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/sink/ConsoleConfigureAction.java
@@ -67,7 +67,13 @@ public class ConsoleConfigureAction {
     }
 
     private static void configureAutoConsole(OutputEventRenderer renderer, ConsoleMetaData consoleMetaData, OutputStream stdout, OutputStream stderr) {
-        if ("windows-aarch64".equals(Platform.current().getId())) {
+        boolean isWindowsAArch64 = false;
+        try { // fixing https://github.com/gradle/gradle/issues/35521 for 9.2.1 to be removed for 9.3.0 because https://github.com/gradle/gradle/pull/35402 makes this obsolete again
+            isWindowsAArch64 = "windows-aarch64".equals(Platform.current().getId());
+        } catch (UnsupportedOperationException unsupportedOperationException) {
+            // unknown os / arch
+        }
+        if (isWindowsAArch64) {
             renderer.addPlainConsole(stdout, stderr);
         } else if (consoleMetaData.isStdOut() && consoleMetaData.isStdErr()) {
             // Redirect stderr to stdout when both stdout and stderr are attached to a console. Assume that they are attached to the same console
@@ -113,7 +119,13 @@ public class ConsoleConfigureAction {
     }
 
     private static void configureRichConsole(OutputEventRenderer renderer, ConsoleMetaData consoleMetaData, OutputStream stdout, OutputStream stderr, boolean verbose) {
-        if ("windows-aarch64".equals(Platform.current().getId())) {
+        boolean isWindowsAArch64 = false;
+        try { // fixing https://github.com/gradle/gradle/issues/35521 for 9.2.1 to be removed for 9.3.0 because https://github.com/gradle/gradle/pull/35402 makes this obsolete again
+            isWindowsAArch64 = "windows-aarch64".equals(Platform.current().getId());
+        } catch (UnsupportedOperationException unsupportedOperationException) {
+            // unknown os / arch
+        }
+        if (isWindowsAArch64) {
             LOGGER.warn("Rich console output is not supported on Windows ARM64. Falling back to plain console output.");
             renderer.addPlainConsole(stdout, stderr);
         } else if (consoleMetaData.isStdOut() && consoleMetaData.isStdErr()) {

